hooks/shotgun_launch_shotgun_app.py

```diff
@@ -26,57 +26,54 @@
 
 HookBaseClass = sgtk.get_hook_baseclass()
 
-class LaunchAssociatedApp(HookBaseClass):
-    def execute(self, path, context, associated_entity, **kwargs):
+
+class LaunchShotgunApp(HookBaseClass):
+    def execute(self, published_file, **kwargs):
         """
         Launches the associated app and starts tank.
-        
+
         :param path: full path to the published file
         :param context: context object representing the publish
         :param associated_entity: same as context.entity
         """
-        status = False
-
-        if context is None:
-            raise TankError("Context cannot be None!")
-
         ########################################################################
         # Example implementation below:
+        path = self.get_publish_path(published_file)
+        valid_photoshop_extensions = [".psd", ".jpg", ".jpeg", ".png", ".tiff", ".tga"]
 
+        if published_file.get("task"):
+            context = self.tank.context_from_entity("Task", published_file["task"].get("id"))
+        else:
+            context = self.tank.context_from_path(path)
```

Comment:
I had previously a comment about catching errors so this app could be used with configs without a file schema?

Reply:
As discussed, will come in another ticket.